packages/flutter_tools/lib/src/compile.dart

```diff
@@ -286,6 +286,7 @@ class KernelCompiler {
       sdkRoot,
       '--strong',
       '--target=$targetModel',
+      '-Ddart.developer.causal_async_stacks=true',
```

Comment:
It seems like we need to decide on causal-async-stacks somewhere up in the call chain, not in the compiler itself. Consider passing a bool argument here.

Reply:
Added this as a parameter to KernelCompiler/ResidentCompiler, but still always using a default value of true since Flutter has (as-yet) no mode that runs with them off.